lib/everest.cpp

```diff
@@ -879,16 +923,30 @@ void Everest::provide_cmd(const std::string& impl_id, const std::string& cmd_nam
                 EVLOG_warning << fmt::format("Ignoring return value of cmd '{}' because the validation of the result "
                                              "failed: {}\ndefinition: {}\ndata: {}",
                                              cmd_name, e.what(), cmd_definition, res_data);
-                return;
+                error = CmdResultError{CmdEvent::SchemaValidationFailed, e.what()};
             }
         }
 
-        EVLOG_verbose << fmt::format("RETVAL: {}", res_data["retval"].dump());
         res_data["origin"] = this->module_id;
+        if (error.has_value()) {
+            res_data["error"] = error.value();
+        }
 
         const json res_publish_data = json::object({{"name", cmd_name}, {"type", "result"}, {"data", res_data}});
 
         this->mqtt_abstraction->publish(cmd_topic, res_publish_data);
+
+        // re-throw exception caught in handler
+        if (error.has_value()) {
+            auto err = error.value();
+            if (err.event == CmdEvent::HandlerException) {
+                if (err.ex) {
+                    std::rethrow_exception(err.ex);
+                } else {
+                    throw std::runtime_error("Unknown exception during cmd handling");
+                }
+            }
+        }
```

Comment:
TODO: should this be configurable? default = rethrow (which is very similar to the current behavior) but can be turned off?

Reply:
For now I would just rethrow it and possibly make it configurable in a subsequent PR after we have discussed and analyzed potential implications of that

Reply:
default is now still the current behavior, but the new behavior can be turned on by setting the following in a config:
```yaml
settings:
  forward_exceptions: true
````